sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -205,6 +206,20 @@ abstract class QueryTest extends PlanTest {
         planWithCaching)
   }
 
+  /**
+   * Asserts that a given [[Dataset]] will be executed using the given named cache.
+   */
+  def assertCached(query: Dataset[_], cachedName: String, storageLevel: StorageLevel): Unit = {
+    val planWithCaching = query.queryExecution.withCachedData
+    val matched = planWithCaching.collectFirst { case cached: InMemoryRelation =>
+      val cacheBuilder = cached.asInstanceOf[InMemoryRelation].cacheBuilder
+      cacheBuilder.tableName.get == cachedName && cacheBuilder.storageLevel == storageLevel
+    }.nonEmpty
```

Comment:
Ur, it looks wrong to me.

Reply:
Hi, @William1104 . Please fix like the following. Otherwise, your test cases will pass without your patches.
```
-    }.nonEmpty
+    }.getOrElse(false)
```

Reply:
At the very beginning it was `planWithCaching.collect {...` . I missed to change `nonEmpty` to `getOrElse(false)` after changing to use `collectFirst` method. 
Just updated. Thanks for pointing out my bug.